catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/FieldDescriptors.js

```diff
@@ -0,0 +1,409 @@
+                        description: 'Number',
+                        type: 'string'
+                    },
+                    extension: {
+                        displayName: 'Extension',
+                        description: 'Extension',
+                        type: 'number'
+                    }
+                },
+                EmailAddress: {
+                    type: {
+                        displayName: 'Email Type',
+                        description: 'Email Type could be work, personal, primary etc.',
+                        values: [],
+                        type: 'string'
+                    },
+                    address: {
+                        displayName: 'Address',
+                        description: 'Email Address',
+                        values: [],
+                        type: 'string'
+                    }
+                },
+                Address: {
+                    street: {
+                        displayName: 'Street',
+                        description: 'Street',
+                        values: [],
+                        type: 'string'
+                    },
+                    city: {
+                        displayName: 'City',
+                        description: 'City',
+                        values: [],
+                        type: 'string'
+                    },
+                    country: {
+                        displayName: 'Country',
+                        description: 'Country',
+                        values: [],
+                        type: 'string'
+                    },
+                    stateOrProvince: {
+                        displayName: 'State or Province',
+                        description: 'State or Province',
+                        values: [],
+                        type: 'string'
+                    },
+                    postalCode: {
+                        displayName: 'Postal Code',
+                        description: 'Postal Code',
+                        values: [],
+                        type: 'string'
+                    }
+                }
+            };
+            if (this.customSlots) {
+                for (var prop in this.customSlots) {
+                    if (this.customSlots.hasOwnProperty(prop)) {
+                        this.mixInCustomSlots(descriptors[prop], this.customSlots[prop]);
+                    }
+                }
+            }
+            return descriptors;
+        },
+        mixInCustomSlots: function (base, custom) {
+            for (var prop in custom) {
+                if (custom.hasOwnProperty(prop)) {
+                    base[prop] = custom[prop];
+                    base[prop].isSlot = true;
+                }
+            }
+        },
+        constructEmailTitle: function () {
+            var title = this.getField('type').get('value') + ' ' + this.getField('address').get('value');
+            title = title.replace(/undefined/g, '').trim();
```

Comment:
I was just trying to make it simpler by not checking every value i put into these title strings and just removing any 'undefined' values afterward. I used this technique in several methods in this file so they should all be consistent on how they do this.